### PR TITLE
[POAE7-2829] fix bugs under C++20

### PR DESCRIPTION
**Pull request by harborn**

### CODE REVIEW

File: cpp/src/cider/util/memory/BufferMgr/CpuBufferMgr/CpuBufferMgr.cpp [hunk 1]
```diff
@@ -36,7 +35,7 @@ void CpuBufferMgr::addSlab(const size_t slab_size) {
   } catch (std::bad_alloc&) {
     slabs_.resize(slabs_.size() - 1);
     CIDER_THROW(CiderOutOfMemoryException,
-                fmt::format("Failed to allocate {}} bytes", slab_size));
+                fmt::format("Failed to allocate {} bytes", slab_size));
```
Comment (Ma-Jian1) [inline cpp/src/cider/util/memory/BufferMgr/CpuBufferMgr/CpuBufferMgr.cpp:38]: use `std::format` in c++20, and remove fmt from CMakeLists.txt and Dockerfile

**Review by Ma-Jian1 — DISMISSED** [review]

File: cpp/src/cider/exec/module/batch/CiderBatch.cpp [hunk 2]
```diff
@@ -21,6 +21,9 @@
 
 #include "cider/batch/CiderBatch.h"
 #include "ArrowABI.h"
+#include "include/cider/batch/ScalarBatch.h"
```
Comment (jikunshang) [inline cpp/src/cider/exec/module/batch/CiderBatch.cpp:24]: `#include "cider/batch/ScalarBatch.h"` is enough?

File: cpp/src/cider/exec/nextgen/context/CMakeLists.txt [hunk 3]
```diff
@@ -19,6 +19,8 @@
 set(CONTEXT_SOURCE
     ${CMAKE_CURRENT_LIST_DIR}/CodegenContext.cpp
     ${CMAKE_CURRENT_LIST_DIR}/RuntimeContext.cpp
-    ${CMAKE_CURRENT_LIST_DIR}/Batch.cpp)
+    ${CMAKE_CURRENT_LIST_DIR}/Batch.cpp
+    ../../../exec/module/batch/CiderBatchUtils.cpp
```
Comment (yma11) [inline cpp/src/cider/exec/nextgen/context/CMakeLists.txt:23]: Better not use relative path in CMake?

File: cpp/src/cider/tests/functionality/CMakeLists.txt [hunk 4]
```diff
@@ -26,6 +26,24 @@ foreach(cider_functionality_test_file ${cider_functionality_test_files})
                ${cider_functionality_test_file})
   string(REGEX REPLACE "functionality/" "" target_name ${test_relative_path})
   string(REGEX REPLACE ".cpp" "" target_name ${target_name})
+  if(${target_name} MATCHES "CiderFilterTest")
```
Comment (jikunshang) [inline cpp/src/cider/tests/functionality/CMakeLists.txt:29]: these files will be removed right? Please add some todo.

**Review by jikunshang — COMMENTED** [review]
overall LGTM.

File: cpp/src/cider/tests/nextgen/functional/CastTest.cpp [hunk 5]
```diff
@@ -42,58 +44,56 @@
...
+    assertQuery(                                                                       \
+        "SELECT CAST(col_a as DOUBLE)  FROM test where CAST(col_b as INTEGER) > 20 "); \
+    assertQuery("SELECT CAST(col_a as INTEGER) + CAST(col_b as INTEGER) FROM test");   \
+    GTEST_SKIP() << "Test skipped since groupby not ready";                            \
+    assertQueryIgnoreOrder(                                                            \
+        "SELECT CAST(col_a as INTEGER), count(col_b) FROM test GROUP BY col_a", "");   \
   }
 
-GEN_CAST_TYPE_TEST_CLASS_FOR_ARROW(Float, Fp32)
+GEN_CAST_TYPE_TEST_CLASS(Float, Fp32)
```
Comment (yma11) [inline cpp/src/cider/tests/nextgen/functional/CastTest.cpp:63]: @kd97 can you help review these update change?

File: cpp/src/cider/tests/utils/CiderProfilingBase.h [hunk 6]
```diff
@@ -31,7 +31,6 @@ class CiderProfilingBase : public CiderTestBase {
  public:
   void benchSQL(const std::string& sql) {
     g_enable_debug_timer = true;
-    LOG(INFO) << "BENCH SQL:" << sql;
```
Comment (yma11) [inline cpp/src/cider/tests/utils/CiderProfilingBase.h:34]: This one should be necessary and should not be removed.

File: cpp/src/cider/tests/nextgen/functional/CastTest.cpp (hunk 5 above)
Comment (kd97) [reply cpp/src/cider/tests/nextgen/functional/CastTest.cpp:63]: better to rebase code? seems no new change in your latest commit? @harborn

File: cpp/src/cider/tests/utils/CiderNextgenQueryRunner.cpp [hunk 7]
```diff
@@ -68,7 +68,6 @@ ::substrait::Plan CiderNextgenQueryRunner::genSubstraitPlan(
 
   ::substrait::Plan plan;
   google::protobuf::util::JsonStringToMessage(json, &plan);
-  LOG(INFO) << "substrait json is: " << json << "\n";
```
Comment (kd97) [inline cpp/src/cider/tests/utils/CiderNextgenQueryRunner.cpp:71]: better to keep this line to print plan info?

**Review by Ma-Jian1 — APPROVED** [review]

**Review by jikunshang — APPROVED** [review]

**Pull request merged**